centraldogma/dogma.py

```diff
@@ -182,3 +193,122 @@ def push(
+        )
+
+    def watch_file(
+        self,
+        project_name: str,
+        repo_name: str,
+        last_known_revision: Revision,
+        query: Query[T],
+        timeout_millis: int = _DEFAULT_WATCH_TIMEOUT_MILLIS,
+    ) -> Optional[Entry[T]]:
+        """
+        Waits for the file matched by the specified ``Query`` to be changed since the specified
+        ``last_known_revision``. If no changes were made within the specified ``timeout_millis``,
+        ``None`` will be returned.
+        It is recommended to specify the largest ``timeout_millis`` allowed by the server. If unsure, use
+        the default watch timeout.
+
+        :return: the ``Entry`` which contains the latest known ``Query`` result.
+                 ``None`` if the file was not changed for ``timeout_millis`` milliseconds
+                 since the invocation of this method.
+        """
+        return self.content_service.watch_file(
+            project_name, repo_name, last_known_revision, query, timeout_millis
+        )
+
+    def repository_watcher(
+        self,
+        project_name: str,
+        repo_name: str,
+        path_pattern: str,
+        function: Callable[[Revision], T] = lambda x: x,
+    ) -> Watcher[T]:
+        """
+        Returns a ``Watcher`` which notifies its listeners when the specified repository has a new commit
+        that contains the changes for the files matched by the given ``path_pattern``. e.g::
+            def get_files(revision: Revision) -> List[Content]:
+                return dogma.get_files("foo_project", "bar_repo", revision, "/*.json")
+
+            with dogma.repository_watcher("foo_project", "bar_repo", "/*.json", get_files) as watcher:
+
+                def listener(revision: Revision, contents: List[Content]) -> None:
+                    ...
+
+                watcher.watch(listener)
+
+        Note that you may get ``RevisionNotFoundException`` during the ``get_files()`` call and
+        may have to retry in the above example due to `a known issue`_.
+
+        :param path_pattern: the path pattern to match files in the repository.
+        :param function: the function to convert the given `Revision` into another.
+
+        .. _a known issue:
+            https://github.com/line/centraldogma/issues/40
+        """
+        watcher = RepositoryWatcher(
+            self.content_service,
+            project_name,
+            repo_name,
+            path_pattern,
+            _DEFAULT_WATCH_TIMEOUT_MILLIS,
+            function,
+        )
+        watcher.start()
+        return watcher
+
+    def file_watcher(
+        self,
+        project_name: str,
+        repo_name: str,
+        query: Query[T],
+        function: Callable[[T], U] = lambda x: x,
+    ) -> Watcher[U]:
+        """
+         Returns a ``Watcher`` which notifies its listeners after applying the specified ``function`` when the result
+         of the given ``Query`` becomes available or changes. e.g::
```

Comment:
```suggestion
        """
        Returns a ``Watcher`` which notifies its listeners after applying the specified ``function`` when the result
        of the given ``Query`` becomes available or changes. e.g::
```
Is it also a convention using `e.g::`(two colons)?

Reply:
I referred to https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html for example code block.
```
 Sphinx's conf.py::

            napoleon_include_special_with_doc = True

```